src/model/Tick.ts

```diff
@@ -42,13 +58,37 @@ export class Tick {
     feeGrowthGlobal1X128: JSBI,
     leftToRight: boolean
   ): boolean {
-    // TODO
-    return false;
+    const liquidityGrossBefore = this.liquidityGross;
+    const liquidityGrossAfter = LiquidityMath.addDelta(
```

Comment:
I'm concerned about balance check, as how can we assert liquidity from a burn request is enough comparing to the one's balance? Contract implementation just do math and assignment to an unsigned integer but we need to check it explicitly. Like `liquidityGrossAfter` here, it should be an unsigned integer, or this can be done in `LiquidityMath.addDelta` which should return unsigned integer. By the way `liquidityNext` in `PositionManager.update` should also ensure this. So we can add assertion here or in the library's method. Which one do you prefer?

Reply:
Also, I wonder whether we should do assertion in the method or just throw an error. Your thoughts are welcomed.

Reply:
It seems to me on the tick level update, it implicitly assumes this check has been done before the call has been made on a higher level. I think we should do the assertion there.

Reply:
ok, I will agree on returning timely, then the check should be done before `ticks.update` and `position.update` in `_updatePosition`. @denice646

Reply:
Let's add the check to both.